SCRWebService/endpoints/tests/test_scr_adaptor_components.py

```diff
@@ -0,0 +1,148 @@
+
+        response = self.fetch(GP_SUMMARY_UPLOAD_URL, method='POST', headers={},
+                              body=json.dumps(body))
+
+        self.assertEqual(response.code, 400)
+        self.assertIn('No interaction-name header provided', response.body.decode())
+
+    @mock.patch('comms.common_https.CommonHttps.make_request')
+    def test_correlation_id_is_passed_to_mhs_if_provided_to_scr_adaptor(self, request_mock):
+        body = file_utilities.FileUtilities.get_file_dict(complete_data_path)
+
+        request_mock.return_value = test_utilities.awaitable("Response message")
+
+        self.fetch(GP_SUMMARY_UPLOAD_URL, method='POST',
+                   headers={
+                       INTERACTION_NAME: GP_SUMMARY_UPLOAD_NAME,
+                       CORRELATION_ID: "123123123"
+                   },
+                   body=json.dumps(body))
+
+        correlation_id_header = request_mock.call_args[1]['headers'][CORRELATION_ID]
+        self.assertEqual(correlation_id_header, "123123123")
+
+    @mock.patch('comms.common_https.CommonHttps.make_request')
+    def test_message_id_is_passed_to_mhs_if_provided_to_scr_adaptor(self, request_mock):
+        body = file_utilities.FileUtilities.get_file_dict(complete_data_path)
+
+        request_mock.return_value = test_utilities.awaitable("Response message")
+
+        self.fetch(GP_SUMMARY_UPLOAD_URL, method='POST',
+                   headers={
+                       INTERACTION_NAME: GP_SUMMARY_UPLOAD_NAME,
+                       MESSAGE_ID: "123123123"
+                   },
+                   body=json.dumps(body))
+
+        correlation_id_header = request_mock.call_args[1]['headers'][MESSAGE_ID]
+        self.assertEqual(correlation_id_header, "123123123")
+
+    @mock.patch('comms.common_https.CommonHttps.make_request')
+    def test_both_message_id_and_correlation_are_passed_to_mhs_when_provided(self, request_mock):
+        body = file_utilities.FileUtilities.get_file_dict(complete_data_path)
+
+        request_mock.return_value = test_utilities.awaitable("Response message")
+
+        self.fetch(GP_SUMMARY_UPLOAD_URL, method='POST',
+                   headers={
+                       INTERACTION_NAME: GP_SUMMARY_UPLOAD_NAME,
+                       MESSAGE_ID: 'messageId1212',
+                       CORRELATION_ID: 'correlationId1212'
+                   },
+                   body=json.dumps(body))
+
+        correlation_id_header = request_mock.call_args[1]['headers']['correlation-id']
+        message_id_header = request_mock.call_args[1]['headers']['message-id']
+        self.assertEqual(message_id_header, "messageId1212")
+        self.assertEqual(correlation_id_header, 'correlationId1212')
+
+    @mock.patch('utilities.message_utilities.MessageUtilities.get_uuid')
+    @mock.patch('comms.common_https.CommonHttps.make_request')
+    def test_when_no_correlation_id_is_provided_the_adaptor_generates_one_and_passes_it_to_the_mhs(self,
```

Comment:
this correlation ID should be returned to the caller in the response headers i would think also ? (so the caller knows what ID is being used to track this request if they were having an issue with the system and needed to report it). This may be happening, but we should assert the response.

Reply:
I believe the intention was to have correlation id as almost a purely internal thing unless they provide it, the MHS also doesn't return the correlation id  which makes me think this shouldn't be added